node-tests/template_compiler_test.js

```diff
@@ -47,144 +48,309 @@ describe('TemplateCompiler', function () {
     htmlbarsPrecompile = htmlbarsOptions.templateCompiler.precompile;
   });
 
-  it(
-    'precompiles templates into htmlbars',
-    co.wrap(function* () {
-      let tree = new TemplateCompiler(input.path(), htmlbarsOptions);
+  describe('requiresModuleApiPolyfill: true', function () {
+    it(
+      'precompiles templates into htmlbars',
+      co.wrap(function* () {
+        let tree = new TemplateCompiler(input.path(), htmlbarsOptions, true);
+
+        output = createBuilder(tree);
+        yield output.build();
+
+        let source = input.readText('template.hbs');
+        let expected = `export default Ember.HTMLBars.template(${htmlbarsPrecompile(source, {
+          moduleName: 'template.hbs',
+        })});`;
+        assert.strictEqual(output.readText('template.js'), expected);
+      })
+    );
+
+    it('invokes AST plugins', async function () {
```

Comment:
thanks for using async/await here, I just haven't had a chance to clean these other co.wrap's up yet